functests/cluster_upgrade_test.go

```diff
@@ -0,0 +1,56 @@
+package functests_test
+
+import (
+	"flag"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	tests "github.com/openshift/ocs-operator/functests"
+
+	deploymanager "github.com/openshift/ocs-operator/pkg/deploy-manager"
+)
+
+var _ = Describe("Cluster upgrade", func() {
+	flag.Parse()
+
+	BeforeEach(func() {
+		RegisterFailHandler(Fail)
+	})
+
+	Describe("ocs", func() {
+
+		AfterEach(func() {
+			By("Uninstall the upgraded cluster")
+			tests.AfterUpgradeTestSuiteCleanup()
+			By("Reinstall a fresh cluster after upgrade")
+			tests.BeforeTestSuiteSetup()
+		})
+
+		Context("upgrade cluster", func() {
+			It("and verify deployment status", func() {
+				if tests.UpgradeToOcsRegistryImage == "" || tests.UpgradeToLocalStorageRegistryImage == "" {
```

Comment:
If we think about when this test will be run, it will typically involve testing new code coming **from** a previous release, rather than new code going **to** a new release.

What would you think about revising the test a bit to deploy an old known working version of the ocs-operator and then updating to the version of the ocs-operator that's being verified in the functional testsuite?

Reply:
I agree that's more intuitive and the standard test pattern. I was going to flip it around once we switched to 0.0.2 as the default environment. I went ahead and made the changes to use UPGRADE_FROM.